core/src/de/tobiasroeser/mill/vcs/version/VcsState.scala

```diff
@@ -24,7 +24,7 @@ case class VcsState(
     val commitCountPart = if (lastTag.isEmpty || commitsSinceLastTag > 0) {
       s"$countSep${if (commitCountPad > 0) {
         (10000000000000L + commitsSinceLastTag).toString().substring(14 - commitCountPad, 14)
-      } else commitsSinceLastTag}"
+      } else if (commitCountPad == 0) commitsSinceLastTag else ""}"
```

Comment:
Even with this change the `revisionPart` will still come after it won't it? So if 

1. tag 0.1.0
2. commit twice
3. wouldn't this produce 0.1.0-SNAPSHOT<revisionPart>

If so, that'd be what I'm trying to avoid. According to sonatype snapshots the version _has_ to end with `-SNAPSHOT` in order to be published.

Reply:
For more context, my goal is this https://github.com/ckipp01/mill-ci-release/actions/workflows/release.yml

Release a snapshot for every merge into main. I'm hacking it now, but the way my versions looks here is what I'd want: `0.1.1-2-16b374-SNAPSHOT` for example. So that every one is unique and ends in `-SNAPSHOT`.

Reply:
OK, then a completely misunderstood your request. To my (probably outdated) knowledge, Maven is handling -SNAPSHOT dependencies specially when deploying to a remote repository, by replacing the SNAPSHOT prefix with a timestamp. 

But now I see, that you don't want to rely on that mechanism, but instead just want to **append** the `-SNAPSHOT`. I think the `-SNAPSHOT` suffix was initially only intended to be used in local repositories, but it looks like it is nowadays commonly used, so yeah, we can also implement that.

I think we still want to keep the essential change of this PR, omitting the count.